MapboxCoreNavigation/RouteController.swift

```diff
@@ -142,11 +157,46 @@ open class RouteController: NSObject {
 }
 
 extension RouteController: CLLocationManagerDelegate {
+    
+    func interpolateLocation() {
+        guard let location = locationManager.lastKnownLocation else { return }
+        guard let polyline = routeProgress.route.coordinates else { return }
+        
+        let distance = location.speed as CLLocationDistance
+        
+        guard let interpolatedCoordinate = coordinate(at: routeProgress.distanceTraveled+distance, fromStartOf: polyline) else {
+            return
+        }
+        
+        var course = location.course
+        if let upcomingCoordinate = coordinate(at: routeProgress.distanceTraveled+(distance*2), fromStartOf: polyline) {
+            course = interpolatedCoordinate.direction(to: upcomingCoordinate)
+        }
+        
+        let interpolatedLocation = CLLocation(coordinate: interpolatedCoordinate,
+                                              altitude: location.altitude,
+                                              horizontalAccuracy: location.horizontalAccuracy,
+                                              verticalAccuracy: location.verticalAccuracy,
+                                              course: course,
+                                              speed: location.speed,
+                                              timestamp: Date())
+        
+        self.locationManager(self.locationManager, didUpdateLocations: [interpolatedLocation])
+    }
+    
     public func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
         guard let location = locations.last else {
             return
         }
         
+        delegate?.routeController!(self, didUpdateLocations: [location])
```

Comment:
Should `routeController(_:didUpdateLocations:)` be optional if we require its existence here?